### PR TITLE
Modify the pubsub example to have custom span.

### PR DESCRIPTION
**Pull request by artursouza**
# Description

Added tracing example to PubSub.
Included command validation.

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #537 

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [X] Code compiles correctly
* [X] Created/updated tests
* [X] Extended the documentation

### CODE REVIEW

**Review by mukundansundar — APPROVED** [review]

File: examples/src/main/java/io/dapr/examples/pubsub/http/README.md [hunk 1]
```diff
@@ -238,6 +238,41 @@ Once running, the Subscriber should print the output as follows:
...
+name: Run Publisher
+expected_stdout_lines:
+  - '== APP == Published message: This is message #0'
+  - '== APP == Published message: This is message #1'
+background: true
+sleep: 15
+-->
+
+```bash
+dapr run --components-path ./components/pubsub --app-id publisher -- java -jar target/dapr-java-sdk-examples-exec.jar io.dapr.examples.pubsub.http.PublisherWithTracing
```
Comment (mukundansundar) [inline examples/src/main/java/io/dapr/examples/pubsub/http/README.md:267]: please change --app-id ... Same app-id cannot exist with dapr run.

**Review by mukundansundar — CHANGES_REQUESTED** [review]

File: examples/src/main/java/io/dapr/examples/pubsub/http/README.md (hunk 1 above)
Comment (artursouza) [reply examples/src/main/java/io/dapr/examples/pubsub/http/README.md:267]: Makes sense.

File: examples/src/main/java/io/dapr/examples/pubsub/http/PublisherWithTracing.java [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+   */
+  public static void main(String[] args) throws Exception {
+    final OpenTelemetry openTelemetry = OpenTelemetryConfig.createOpenTelemetry();
+    final Tracer tracer = openTelemetry.getTracer(PublisherWithTracing.class.getCanonicalName());
+    Span span = tracer.spanBuilder("Publisher's Main").setSpanKind(Span.Kind.CLIENT).startSpan();
+
+    try (DaprClient client = new DaprClientBuilder().build()) {
+      try (Scope scope = span.makeCurrent()) {
+        for (int i = 0; i < NUM_MESSAGES; i++) {
+          String message = String.format("This is message #%d", i);
```
[the remark this answers is not in the document]
Comment (artursouza) [reply examples/src/main/java/io/dapr/examples/pubsub/http/PublisherWithTracing.java:53]: Makes sense.

**Review by halspang — COMMENTED** [review]
Seems like the examples are failing to build while they're fine on master. Not sure how this change could've broken them but is that something we should check out?

**Comment by artursouza** [comment]
I will take a look.

**Comment by artursouza** [comment]
> I will take a look.

For some reason, it was doing findbug checks on examples. Which did not do after I reran it. Locally, I had the same experience but it worked after I did a mvn clean.

**Review by mukundansundar — APPROVED** [review]

**Pull request merged**